pkg/discovery/discovery.go

```diff
@@ -183,6 +191,32 @@ func Run(restConf *rest.Config, cfg *config.Config) (errCount int) {
 	return errCount
 }
 
+// For the namespaces specified by cfg.Limits.PodLogs.Namespaces and cfg.Limits.PodLogs.SonobuoyNamespace,
+// transform to cfg.Limits.PodLogs.FieldSelectors[
+//     ...,
+//     "metadata.namespace=xxx1",
+//     "metadata.namespace=xxx2"
+// ], which acts as filters when gathering pod logs
+func preprocessPodLogLimits(cfg *config.Config, kubeClient *kubernetes.Clientset) {
+	var nsfilter string
+	if len(cfg.Limits.PodLogs.Namespaces) > 0 {
+		nsfilter = cfg.Limits.PodLogs.Namespaces
+	}
+	if cfg.Limits.PodLogs.SonobuoyNamespace != nil && *cfg.Limits.PodLogs.SonobuoyNamespace {
+		if len(nsfilter) > 0 {
+			nsfilter = fmt.Sprintf("%s|%s", nsfilter, cfg.Namespace)
+		} else {
+			nsfilter = cfg.Namespace
+		}
+	}
+	if len(nsfilter) > 0 {
+		nsListLogs, _ := FilterNamespaces(kubeClient, nsfilter)
+		for _, ns := range nsListLogs {
+			cfg.Limits.PodLogs.FieldSelectors = append(cfg.Limits.PodLogs.FieldSelectors, "metadata.namespace="+ns)
```

Comment:
Let's not have a method which will mutate the cfg like this.
 - it makes the signature confusing since you don't know what it really intends to do to the object
 - it will make it permanently unclear if a config object you have is ready to be used or if you need to call this method to get it ready.

Instead, I'd take this method and have it return a list of field selectors to run queries on.

Reply:
In addition, this method should definitely have a set of tests to cover it.

I don't think it needs to take the `kubeClient` either. It seems like we should just be able to do a few things:
 - given a config, get the namespace filter (a combination of Namespaces and SonobuoyNamespace)
 - given the config object and a list of namespaces, generate a list of field selectors to query

Will comment at the calling site about this too to try and make it clear.

Reply:
The method could take input of config, and output a regex for the targeted namespaces